### PR TITLE
Add Console.AutoCompleteMode variable

### PR DESCRIPTION
**Pull request by ThiconZ**
### Proposed changes in this pull request:

1. Add Console.AutoCompleteMode variable

### Why should this pull request be merged?

Many games that contain a console will provide players with a choice of Prefix or Substring based auto completion. This adds in the option to allow players to pick what style they want - with it defaulting to Prefix.

### Have you tested this on your own and/or in a game with other people?

Yes.

### CODE REVIEW

**Comment by camden-smallwood** [comment]
Instead of using an int value, it'd be more ideal to use an enum or an enum class to store the auto-complete mode.
Example:
```Cpp
enum class AutoCompleteMode : int
{
    Prefix,
    Substring
};
```

**Comment by ThiconZ** [comment]
@camden-smallwood I added an enum-style variable and switched all instances to using it.

**Comment by maximumgame** [comment]
It seems selecting any mode results in toggling rather than just 0 being a toggle.
o9cpmkr

**Comment by camden-smallwood** [comment]
Looks pretty good so far, but I think you should add string support for the auto-complete mode on top of the integer value, like:
```
Console.AutoCompleteMode partial
Console.AutoCompleteMode substring
```
Then, maybe add a separate toggling command?
```
Console.ToggleAutoCompleteMode
```
Because I feel that running the command with no arguments should return the current value of the var:
```
Console.AutoCompleteMode
> partial
````

File: dist/mods/ui/web/screens/console/console.js [hunk 1]
```diff
@@ -718,19 +782,3 @@ $(window).load(function () {
...
-function formatBytes(bytes,decimals) {
-    if(bytes == 0) return '0 Bytes';
-    var k = 1000,
-        dm = decimals || 2,
-        sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'],
-        i = Math.floor(Math.log(bytes) / Math.log(k));
-    return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
-}
-
-function showMemoryUsage() {
```
Comment (medsouz) [inline dist/mods/ui/web/screens/console/console.js]: Why did you remove this code?

Comment (ThiconZ) [inline dist/mods/ui/web/screens/console/console.js]: I guess it was deleted when I sync'd. I'll add it back in the next commit.

**Comment by ThiconZ** [comment]
I fixed the two issues of the functions being missing and it always doing a toggle. It now also allows taking in a string value though it does not currently inform the player what string values are accepted nor does it print out the current value as a string.

**Comment by ThiconZ** [comment]
I've added all the string names to the output/help text. This is how it looks @camden-smallwood 
image
